catroid/src/main/java/org/catrobat/catroid/ProjectManager.java

```diff
@@ -167,9 +167,13 @@ public void loadProject(File projectDir, Context context) throws ProjectExceptio
 		if (project.getCatrobatLanguageVersion() <= 0.9999995) {
 			updateBackgroundIndexTo9999995(project);
 		}
+		if (project.getCatrobatLanguageVersion() < 1.01 && BuildConfig.DEBUG) {
```

Comment:
```suggestion
		if (project.getCatrobatLanguageVersion() < 999.9 && BuildConfig.DEBUG) {
		// Delete the DEBUG flag and replace 999.9 by the newest Catrobat language version number 
		// once https://jira.catrob.at/browse/CATROID-851 has been implemented and merged.
```
The version number in the condition here will have to be set when the transition to lists as primitive data structures is made. 

Note to myself: The current Google Play version supports CatrobatLanguageVersion 1.03, so I tested it with an earlier version, and the insertion worked fine. Test project to check for insertion was https://share.catrob.at/app/project/c4b5c4ed-c616-11eb-abc2-005056a36f47 (this project is only visible with debug builds).

Reply:
The check "< 999.9" would always be true and the way it is implemented now, flattenAllLists would be called at every LoadProject. Then Flatten would be inserted again and again.

The version check prevents this at the moment. Flatten is added once when an old project is loaded. Then the version number of the project is updated and next time Flatten is not executed again.

Perhaps it would be better to set the version number for comparison to 0 and NOT execute flattenAllLists at the moment and only include this functionality when we know the language version number of "lists as primitive data structure"?

Reply:
Sorry for the late reply. Inserting the flatten function again and again is no problem, as long as this is changed to the actual version number required before the release with the full feature with the "lists as primitive data structure", since we use the debug versions for testing only. Since it is for testing only, inserting again and again is no problem. If we keep 1.01 or 0 here, we cannot test the flatten functionality with the debug version, since we cannot easily create projects with the old version number or even earlier ones. And this line will anyway have to be changed once the feature can be released.